### PR TITLE
[HOTFIX] optimize module dependency

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
 there are many redundancy module dependencies
 
 ### What changes were proposed in this PR?
1.  optimize module dependency
2.  fix HiveExample testcase
3.  avoid to getOrCreate sparkSession in testcase again
4. add more testcases
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/580/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2285/

File: integration/spark/src/main/scala/org/apache/spark/sql/EnvHelper.scala [hunk 1]
```diff
@@ -26,17 +28,29 @@ object EnvHelper {
 
   def isCloud(sparkSession: SparkSession): Boolean = false
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/spark/sql/EnvHelper.scala]: please rename to a more intuitive name

File: integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.apache.spark.sql.parser
```
Comment (jackylk) [inline integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala]: missing license header

File: integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.apache.spark.sql.catalyst.parser.SqlBaseParser._
+import org.apache.spark.sql.execution.SparkSqlAstBuilder
+import org.apache.spark.sql.internal.SQLConf
+
+class SparkSqlAstBuilderWrapper(conf: SQLConf)
```
Comment (jackylk) [inline integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala]: please add comment

File: sdk/sdk/pom.xml [hunk 4]
```diff
@@ -44,7 +44,6 @@
     <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-aws</artifactId>
-      <version>${hadoop.version}</version>
```
Comment (jackylk) [inline sdk/sdk/pom.xml]: please remove "build-all" profile also as we build all modules now

File: integration/spark/src/main/scala/org/apache/spark/sql/EnvHelper.scala (hunk 1 above)
Comment (QiangCai) [reply integration/spark/src/main/scala/org/apache/spark/sql/EnvHelper.scala]: already change to isLegacy

File: integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala (hunk 2 above)
Comment (QiangCai) [reply integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala]: fixed

File: integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala (hunk 3 above)
Comment (QiangCai) [reply integration/spark/src/main/spark2.4/org/apache/spark/sql/parser/SparkSqlAstBuilderWrapper.scala]: added

File: sdk/sdk/pom.xml (hunk 4 above)
Comment (QiangCai) [reply sdk/sdk/pom.xml]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/592/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2299/

**Comment by jackylk** [comment]
LGTM

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/cloud/CacheRefreshTestCase.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package org.apache.carbondata.spark.testsuite.cloud
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/cloud/CacheRefreshTestCase.scala]: Please Add License

File: integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOpName.scala [hunk 6]
```diff
@@ -2079,6 +2079,9 @@ class TestStreamingTableOpName extends QueryTest with BeforeAndAfterAll {
       Thread.sleep(continueSeconds * 1000)
       thread2.interrupt()
       thread1.interrupt()
+    } catch {
+      case ex =>
+        LOGGER.error("finished to ingest data", ex)
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableOpName.scala]: Please change the error message, as it represents a failure

File: integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableQueryFilter.scala [hunk 7]
```diff
@@ -240,6 +240,9 @@ class TestStreamingTableQueryFilter extends QueryTest with BeforeAndAfterAll {
       Thread.sleep(continueSeconds * 1000)
       thread2.interrupt()
       thread1.interrupt()
+    } catch {
+      case ex =>
+        LOGGER.error("finished to ingest data", ex)
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableQueryFilter.scala]: Please change the error message, as it represents a failure

File: integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableWithLongString.scala [hunk 8]
```diff
@@ -478,6 +478,9 @@ class TestStreamingTableWithLongString extends QueryTest with BeforeAndAfterAll
       Thread.sleep(continueSeconds * 1000)
       thread2.interrupt()
       thread1.interrupt()
+    } catch {
+      case ex =>
+        LOGGER.error("finished to ingest data", ex)
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableWithLongString.scala]: Please change the error message, as it represents a failure

File: integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableWithRowParser.scala [hunk 9]
```diff
@@ -860,6 +860,9 @@ class TestStreamingTableWithRowParser extends QueryTest with BeforeAndAfterAll {
       Thread.sleep(continueSeconds * 1000)
       thread2.interrupt()
       thread1.interrupt()
+    } catch {
+      case ex =>
+        LOGGER.error("finished to ingest data", ex)
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/TestStreamingTableWithRowParser.scala]: Please change the error message, as it represents a failure

**Pull request closed without merging**